src/java.base/windows/classes/sun/net/www/protocol/jar/JarFileFactory.java

```diff
@@ -259,4 +266,29 @@ private Permission getPermission(JarFile jarFile) {
 
         return null;
     }
+
+    @Override
+    public void beforeCheckpoint(Context<? extends Resource> context) throws Exception {
```

Comment:
It seems strange windows has it's own JarFileFactory, mostly duplicating a similar one. I'm fine with the current implementation, but it also seems possible to refactor and extract a single common resource implementation for windows and linux.

Reply:
Could we do it after JDK version upgrade?